DOCUMENTATION.md

```diff
@@ -43,6 +45,190 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
+
+### Supported Inputs
+
+Propulsion provides for processing events from the following sources:
+
+1. `Propulsion.CosmosStore`: Lightly wraps the Change Feed Processor library within the `Microsoft.Azure.Cosmos` client. It provides a load balanced leasing system (like Kafka), enabling multiple instances of a processor to parallelise processing. It allows to parallelise processing for multiple consumers. It's done by using [competing consumers pattern](https://www.enterpriseintegrationpatterns.com/patterns/messaging/CompetingConsumers.html) with the same Processor Name (Consumer Group Name). There can be as many active processors as physical partitions. The client API indicates the Partition Identity via the Context's `LeaseToken`).
+   - Each physical Partition maps to a Tranche, so processing still honors stream concurrency limits, even if the CFP library assigns more than one partition to a given instance.
+   - The CFP library provides for estimation of the Lag of documents to be processed between the current checkpoint position and the tail of each partition, which can surfaced as a metric.
+   - Over time, as a Container splits into more physical partitions (either due to assigning > 10K RU per physical partition, or by exceeding the 50GB/partition limit), more Partitions will arise.
+   - Checkpoints are maintained by the CFP library within an auxiliary (by convention, a sister container with the name`{container}-aux`) CosmosDB container (the document containing the checkpoint doubles as a lease management target)
+2. `Propulsion.DynamoStore`: Reads from an Index Table maintained by a `Propulsion.DynamoStore.Indexer` Lambda
+    - In the current implementation, the Indexer feeds all items into a single Tranche within its index (extending it to support multiple Tranches based on a hash of the stream identifier would not be difficult).
+    - If an index was to present multiple tranches, the Reader is ready to handle them as is (although there's no support within the source for load balancing (splitting reads by Stream Name hash and then assigning shards across all active instances) across multiple processor instances).
+3. `Propulsion.EventStoreDb`: Uses the EventStoreDb gRPC based API to pull events from the `$all` stream.
+   - In the current implementation, there's no support within the source for load balancing (splitting reads by Stream Name hash and then assigning shards across all active instances) across multiple processor instances
+   - In the current implementation, there's no support for surfacing lag metrics on a continual basis (the reader does report it initially)
+   - There are facilities for storing checkpoints in CosmosStore, DynamoStore, Postgres, SQL Server. There is not presently a [checkpoint store implementation that maintains the checkpoints EventStoreDb itself at present](https://github.com/jet/propulsion/issues/8).
```

Comment:
```suggestion
   -  Propulsion provides you out of the box checkpoint storage in CosmosStore, DynamoStore, Postgres, SQL Server. There is not presently a [checkpoint store implementation that maintains the checkpoints EventStoreDb itself at present](https://github.com/jet/propulsion/issues/8).
```

Regarding checkpoint store in ESDB in itself, you can steal it from: https://github.com/oskardudycz/EventSourcing.NetCore/blob/main/Core.EventStoreDB/Subscriptions/EventStoreDBSubscriptionCheckpointRepository.cs 🙂

Reply:
I know, but will wait for someone else to do that, who can verify it works 100% in prod 
I have an impl in https://github.com/jet/dotnet-templates/pull/81
and then it needs the metadata